ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -147,6 +147,9 @@ - (void)updateViewControllerIfNeeded
   // if nav is nil, it means we can be in a fullScreen modal, so there is no nextVC, but we still want to update
   if (vc != nil && (nextVC == vc || isInFullScreenModal || isPresentingVC)) {
     [RNSScreenStackHeaderConfig updateViewController:self.screenView.controller withConfig:self animated:YES];
+    // As the header might have change in `updateViewController` we need to ensure that header height
+    // returned by the `onHeaderHeightChange` event is correct.
+    [(RNSScreen *)self.screenView.controller calculateAndNotifyHeaderHeightChangeIsModal:NO];
```

Comment:
The cast is still here

```suggestion
    [self.screenView.controller calculateAndNotifyHeaderHeightChangeIsModal:NO];
```

I believe it is redundant

Reply:
Yeah, I've left it on mistake, as I was in a rush :sweat_smile: Fixed in https://github.com/software-mansion/react-native-screens/pull/1917/commits/40df2fdab3cfd753dc7263246515ca6d449e0321.